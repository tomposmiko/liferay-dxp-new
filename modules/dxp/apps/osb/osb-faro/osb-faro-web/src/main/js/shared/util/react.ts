import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
