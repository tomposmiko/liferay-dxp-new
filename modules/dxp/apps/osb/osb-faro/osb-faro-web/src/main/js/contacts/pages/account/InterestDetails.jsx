import {} from 'shared/util/router';
import BaseInterestDetails from '../BaseInterestDetails';
import PropTypes from 'prop-types';
import React from 'react';
import {Account} from 'shared/util/records';
import {ACCOUNTS, INDIVIDUALS, Routes} from 'shared/util/router';

export default class InterestDetails extends React.Component {
	static propTypes = {
		account: PropTypes.instanceOf(Account).isRequired
	};

	render() {
		const {account, tabId = INDIVIDUALS, ...otherProps} = this.props;

		return (
			<BaseInterestDetails
				{...otherProps}
				entity={account}
				interestDetailsRoute={Routes.CONTACTS_ACCOUNT_INTEREST_DETAILS}
				tabId={tabId}
				type={ACCOUNTS}
			/>
		);
	}
}
