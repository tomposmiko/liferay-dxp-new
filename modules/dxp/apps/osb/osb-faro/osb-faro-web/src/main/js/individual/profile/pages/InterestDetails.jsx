import Card from 'shared/components/Card';
import ClayIcon from '@clayui/icon';
import ClayLink from '@clayui/link';
import InterestPagesList from 'contacts/components/InterestPagesList';
import Nav from 'shared/components/Nav';
import React from 'react';
import {EntityTypes} from 'shared/util/constants';
import {Individual} from 'shared/util/records';
import {PropTypes} from 'prop-types';
import {Routes, toRoute} from 'shared/util/router';
import {setUriQueryValue} from 'shared/util/router';
import {sub} from 'shared/util/lang';

export default class InterestDetails extends React.Component {
	static defaultProps = {
		active: 'true'
	};

	static propTypes = {
		active: PropTypes.string,
		channelId: PropTypes.string,
		groupId: PropTypes.string.isRequired,
		id: PropTypes.string.isRequired,
		individual: PropTypes.instanceOf(Individual).isRequired,
		interestId: PropTypes.string.isRequired
	};

	getNavigationItems() {
		const {channelId, groupId, id, interestId} = this.props;

		const active = this.props.active === 'true';

		return [
			{
				active,
				href: setUriQueryValue(
					toRoute(Routes.CONTACTS_INDIVIDUAL_INTEREST_DETAILS, {
						channelId,
						groupId,
						id,
						interestId
					}),
					'active',
					true
				),
				label: Liferay.Language.get('active-pages')
			},
			{
				active: !active,
				href: setUriQueryValue(
					toRoute(Routes.CONTACTS_INDIVIDUAL_INTEREST_DETAILS, {
						channelId,
						groupId,
						id,
						interestId
					}),
					'active',
					false
				),
				label: Liferay.Language.get('inactive-pages')
			}
		];
	}

	render() {
		const {
			props: {channelId, groupId, id, interestId}
		} = this;

		const interestName = decodeURIComponent(interestId);

		return (
			<>
				<div className='back-button-root mb-2'>
					<ClayLink
						borderless
						button
						displayType='secondary'
						href={toRoute(Routes.CONTACTS_INDIVIDUAL_INTERESTS, {
							channelId,
							groupId,
							id
						})}
					>
						<ClayIcon
							className='icon-root mr-2'
							symbol='angle-left'
						/>

						{Liferay.Language.get('back-to-interests')}
					</ClayLink>
				</div>

				<Card pageDisplay>
					<Card.Header>
						<Card.Title>
							{sub(
								Liferay.Language.get('pages-containing-x'),
								[
									<span
										className='interest-name'
										key='INTEREST_NAME'
									>
										{`"${interestName}"`}
									</span>
								],
								false
							)}
						</Card.Title>
					</Card.Header>

					<Card.Header>
						<Nav
							className='page-subnav'
							display='underline'
							key='subnav'
						>
							{this.getNavigationItems().map(
								({active, href, label}) => (
									<Nav.Item
										active={active}
										href={href}
										key={label}
									>
										<h4>{label}</h4>
									</Nav.Item>
								)
							)}
						</Nav>
					</Card.Header>

					<InterestPagesList
						channelId={channelId}
						className='interest-history-table'
						dataSourceParams={{
							active: this.props.active === 'true',
							contactsEntityId: id,
							contactsEntityType: EntityTypes.Individual,
							groupId,
							interestName
						}}
						groupId={groupId}
					/>
				</Card>
			</>
		);
	}
}
