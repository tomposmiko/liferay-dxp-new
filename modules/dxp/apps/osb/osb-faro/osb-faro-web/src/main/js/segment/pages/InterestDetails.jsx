import BaseInterestDetails from 'contacts/pages/BaseInterestDetails';
import PropTypes from 'prop-types';
import React from 'react';
import {Routes} from 'shared/util/router';
import {Segment} from 'shared/util/records';
import {SEGMENTS} from 'shared/util/router';

export default class InterestDetails extends React.Component {
	static propTypes = {
		segment: PropTypes.instanceOf(Segment).isRequired
	};

	render() {
		const {segment, ...otherProps} = this.props;

		return (
			<BaseInterestDetails
				{...otherProps}
				entity={segment}
				interestDetailsRoute={Routes.CONTACTS_SEGMENT_INTEREST_DETAILS}
				type={SEGMENTS}
			/>
		);
	}
}
