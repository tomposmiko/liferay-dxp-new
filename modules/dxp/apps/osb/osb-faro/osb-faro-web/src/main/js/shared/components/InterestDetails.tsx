import Card from 'shared/components/Card';
import getCN from 'classnames';
import getMetricsMapper from 'shared/hoc/mappers/metrics';
import React from 'react';
import TouchpointsQuery from 'shared/queries/TouchpointsQuery';
import URLConstants from 'shared/util/url-constants';
import {createOrderIOMap, VISITORS_METRIC} from 'shared/util/pagination';
import {graphql} from '@apollo/react-hoc';
import {
	metricsListColumns,
	sitePagesListColumns
} from 'shared/util/table-columns';
import {OrderedMap} from 'immutable';
import {OrderParams} from 'shared/util/records';
import {RangeSelectors, Router} from 'shared/types';
import {Routes} from 'shared/util/router';
import {sub} from 'shared/util/lang';
import {useParams} from 'react-router-dom';
import {useQueryPagination, useQueryRangeSelectors} from 'shared/hooks';
import {withBaseResults, withRangeKey} from 'shared/hoc';

const withData = () =>
	graphql(
		TouchpointsQuery,
		getMetricsMapper(result => ({
			items: result.pages.assetMetrics,
			total: result.pages.total
		}))
	);

interface ITableWithDataProps {
	channelId: string;
	delta: number;
	groupId: string;
	interestId: string;
	orderIOMap: OrderedMap<string, OrderParams>;
	page: number;
	query: string;
	rangeSelectors: RangeSelectors;
	router: Router;
}

const TableWithData: React.FC<ITableWithDataProps> = withRangeKey(
	withBaseResults(withData, {
		emptyDescription: sub(
			Liferay.Language.get('empty-message-lists'),
			[
				<a
					href={URLConstants.DocumentationLink}
					key='DOCUMENTATION'
					target='_blank'
				>
					{Liferay.Language.get('documentation').toLowerCase()}
				</a>
			],
			false
		),
		emptyTitle: Liferay.Language.get('empty-title-pages'),
		getColumns: ({channelId, groupId, rangeSelectors}) => [
			sitePagesListColumns.getTitleUrl({
				channelId,
				groupId,
				rangeSelectors,
				route: Routes.SITES_TOUCHPOINTS_OVERVIEW
			}),
			metricsListColumns.visitorsMetric,
			metricsListColumns.viewsMetric,
			metricsListColumns.avgTimeOnPageMetric,
			metricsListColumns.bounceRateMetric,
			metricsListColumns.entrancesMetric,
			metricsListColumns.exitRateMetric
		],
		legacyDropdownRangeKey: false,
		rowIdentifier: 'assetId',
		showDropdownRangeKey: true
	})
);

interface IInterestDetailsProps {
	className?: string;
	router: Router;
}

const InterestDetails: React.FC<IInterestDetailsProps> = ({
	className,
	router
}) => {
	const {channelId, groupId, interestId} = useParams();
	const {delta, orderIOMap, page, query} = useQueryPagination({
		initialOrderIOMap: createOrderIOMap(VISITORS_METRIC)
	});

	const rangeSelectors = useQueryRangeSelectors();

	return (
		<Card className={getCN(className)} pageDisplay>
			<Card.Header className='align-items-center d-flex justify-content-between'>
				<Card.Title>
					{sub(
						Liferay.Language.get('pages-containing-x'),
						[
							<span className='interest-title' key='INTEREST_ID'>
								{`"${interestId}"`}
							</span>
						],
						false
					)}
				</Card.Title>
			</Card.Header>

			<TableWithData
				channelId={channelId}
				delta={delta}
				groupId={groupId}
				interestId={interestId}
				orderIOMap={orderIOMap}
				page={page}
				query={query}
				rangeSelectors={rangeSelectors}
				router={router}
			/>
		</Card>
	);
};

export default InterestDetails;
