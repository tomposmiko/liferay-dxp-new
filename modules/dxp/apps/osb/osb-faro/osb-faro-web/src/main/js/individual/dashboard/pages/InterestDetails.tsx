import BackButton from 'contacts/components/BackButton';
import BasePage from 'shared/components/base-page';
import InterestDetails from 'shared/components/InterestDetails';
import React from 'react';
import {isNil, pickBy} from 'lodash';
import {Router} from 'shared/types';
import {Routes, setUriQueryValues, toRoute} from 'shared/util/router';

interface IInterestDetailsProps extends React.HTMLAttributes<HTMLDivElement> {
	router: Router;
}

const InterestDetailsPage: React.FC<IInterestDetailsProps> = ({router}) => {
	const {
		params: {channelId, groupId},
		query: {rangeKey}
	} = router;

	return (
		<BasePage.Body
			className='individuals-dashboard-interest-details-root'
			pageContainer
		>
			<BackButton
				href={setUriQueryValues(
					pickBy({rangeKey}, param => !isNil(param)),

					toRoute(Routes.CONTACTS_INDIVIDUALS_INTERESTS, {
						channelId,
						groupId
					})
				)}
				label={Liferay.Language.get('back-to-interests')}
			/>

			<InterestDetails router={router} />
		</BasePage.Body>
	);
};

export default InterestDetailsPage;
