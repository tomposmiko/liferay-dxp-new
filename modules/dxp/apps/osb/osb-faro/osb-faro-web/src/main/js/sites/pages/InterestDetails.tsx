import ClayIcon from '@clayui/icon';
import ClayLink from '@clayui/link';
import InterestDetails from 'shared/components/InterestDetails';
import React from 'react';
import {getRangeSelectorsFromQuery} from 'shared/util/util';
import {pickBy} from 'lodash';
import {Router} from 'shared/types';
import {Routes, setUriQueryValues, toRoute} from 'shared/util/router';

interface IInterestDetailsProps extends React.HTMLAttributes<HTMLDivElement> {
	router: Router;
}

export default class InterestDetailsPage extends React.Component<IInterestDetailsProps> {
	render() {
		const {router} = this.props;

		const {
			params: {channelId, groupId},
			query
		} = router;

		const rangeSelectors = getRangeSelectorsFromQuery(query);

		return (
			<div className='sites-dashboard-interest-details-root'>
				<div className='row'>
					<div className='col-xl-12'>
						<div className='back-button-root mb-2'>
							<ClayLink
								borderless
								button
								displayType='secondary'
								href={setUriQueryValues(
									pickBy({...rangeSelectors}),

									toRoute(Routes.SITES_INTERESTS, {
										channelId,
										groupId
									})
								)}
							>
								<ClayIcon
									className='icon-root mr-2'
									symbol='angle-left'
								/>

								{Liferay.Language.get('back-to-interests')}
							</ClayLink>
						</div>

						<InterestDetails
							className='sites-interest-details-root'
							router={router}
						/>
					</div>
				</div>
			</div>
		);
	}
}
