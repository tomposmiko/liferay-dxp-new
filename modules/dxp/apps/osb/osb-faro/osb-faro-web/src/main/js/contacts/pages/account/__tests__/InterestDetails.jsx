import InterestDetails from '../InterestDetails';
import mockDate from 'test/mock-date';
import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {cleanup, render} from '@testing-library/react';

jest.unmock('react-dom');

describe('InterestDetails', () => {
	beforeAll(() => mockDate());
	afterAll(() => jest.restoreMocks());

	afterEach(cleanup);

	it('should render', () => {
		const {container} = render(
			<BrowserRouter>
				<InterestDetails
					account={{}}
					groupId='123'
					id='123'
					interestId='123'
				/>
			</BrowserRouter>
		);

		expect(container).toMatchSnapshot();
	});
});
