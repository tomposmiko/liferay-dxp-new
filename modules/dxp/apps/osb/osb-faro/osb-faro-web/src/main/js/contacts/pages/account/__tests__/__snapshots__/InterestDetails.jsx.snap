// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestDetails should render 1`] = `
<div>
  <div
    class="interest-details-root"
  >
    <div
      class="back-button-root mb-2"
    >
      <a
        class="btn btn-outline-borderless btn-outline-secondary"
        href="/workspace/123/contacts/accounts/123/interests"
      >
        <svg
          class="lexicon-icon lexicon-icon-angle-left icon-root icon-size-sm mr-2"
          role="presentation"
        >
          <use
            xlink:href="#angle-left"
          />
        </svg>
        Back to Interests
      </a>
    </div>
    <div
      class="card card-root page-display"
    >
      <div
        class="card-header"
      >
        <h5
          class="card-title"
        >
          Interest - 
          <span
            class="interest-name"
          >
            123
          </span>
        </h5>
      </div>
      <div
        class="card-header"
      >
        <ul
          class="nav-root page-subnav nav nav-underline"
          role="tablist"
        >
          <li
            class="nav-item active"
            role="presentation"
          >
            <a
              class="nav-link active"
              href="/workspace/123/contacts/accounts/123/interests/123/individuals"
              role="tab"
            >
              <h4>
                Individuals
              </h4>
            </a>
          </li>
          <li
            class="nav-item"
            role="presentation"
          >
            <a
              class="nav-link"
              href="/workspace/123/contacts/accounts/123/interests/123/pages?active=true"
              role="tab"
            >
              <h4>
                Active Pages
              </h4>
            </a>
          </li>
          <li
            class="nav-item"
            role="presentation"
          >
            <a
              class="nav-link"
              href="/workspace/123/contacts/accounts/123/interests/123/pages?active=false"
              role="tab"
            >
              <h4>
                Inactive Pages
              </h4>
            </a>
          </li>
        </ul>
        <h4
          class="list-title"
        >
          Members Interested in "123" - As of January 1, 1970
        </h4>
      </div>
      <div
        class="base-results-root d-flex flex-column flex-grow-1 searchable-table-root interest-history-table d-flex flex-column flex-grow-1 searchable-table-borderless"
      >
        <div
          class="toolbar-root"
        >
          <nav
            class="navbar navbar-root management-bar management-bar-light navbar-expand"
          >
            <div
              class="container-fluid"
            >
              <ul
                class="nav-root front-nav navbar-nav"
                role="tablist"
              >
                <li
                  class="nav-item"
                  role="presentation"
                >
                  <div
                    class="dropdown dropdown-root filter-and-order-root"
                  >
                    <button
                      aria-controls="clay-dropdown-menu-1"
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="dropdown-toggle d-flex align-items-center btn btn-outline-borderless btn-sm btn-outline-secondary"
                      type="button"
                    >
                      Order
                      <svg
                        class="lexicon-icon lexicon-icon-caret-bottom ml-2 mt-0"
                        role="presentation"
                      >
                        <use
                          xlink:href="#caret-bottom"
                        />
                      </svg>
                    </button>
                  </div>
                </li>
                <li
                  class="nav-item"
                  role="presentation"
                >
                  <a
                    class="btn-root nav-link nav-link-monospaced btn btn-outline-borderless btn-outline-unstyled"
                    href="/?field=name&page=1&sortOrder=DESC"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-order-arrow-ascending icon-root"
                      role="presentation"
                    >
                      <use
                        xlink:href="#order-arrow-ascending"
                      />
                    </svg>
                  </a>
                </li>
              </ul>
              <div
                class="navbar-form navbar-form-autofit"
              >
                <div
                  class="input-group search"
                >
                  <div
                    class="input-group-item"
                  >
                    <input
                      class="input-root form-control input-group-inset input-group-inset-after"
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                    <div
                      class="input-group-inset-item input-group-inset-item-after"
                    >
                      <button
                        class="button-root btn btn-unstyled"
                        type="button"
                      >
                        <svg
                          class="lexicon-icon lexicon-icon-search icon-root"
                          role="presentation"
                        >
                          <use
                            xlink:href="#search"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="table-responsive table-root flex-grow-1"
        >
          <table
            class="table table-autofit table-hover table-heading-nowrap table-nowrap"
          >
            <thead>
              <tr>
                <th
                  class="table-head-title"
                >
                  <a
                    class="button-root inline-item text-truncate-inline btn btn-unstyled"
                    href="/?field=name&page=1&sortOrder=DESC"
                  >
                    <span
                      class="text-truncate"
                    >
                      Name
                    </span>
                    <span
                      class="inline-item inline-item-after"
                    >
                      <svg
                        class="lexicon-icon lexicon-icon-order-arrow-up icon-root"
                        role="presentation"
                      >
                        <use
                          xlink:href="#order-arrow-up"
                        />
                      </svg>
                    </span>
                  </a>
                </th>
                <th
                  class="table-head-title"
                >
                  Email
                </th>
                <th
                  class="table-head-title"
                >
                  Account Names
                </th>
              </tr>
            </thead>
          </table>
          <div
            class="spinner-root spinner-overlay"
          >
            <span
              class="loading-animation loading-animation-dotted"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
